nemo/collections/tts/modules/fastpitch.py

```diff
@@ -288,11 +303,18 @@ def infer(self, *, text, pitch=None, speaker=None, pace=1.0, volume=None):
         enc_out, enc_mask = self.encoder(input=text, conditioning=spk_emb)
 
         # Predict duration and pitch
-        log_durs_predicted = self.duration_predictor(enc_out, enc_mask)
+        if self.speaker_emb_condition_prosody:
+            log_durs_predicted = self.duration_predictor(enc_out + spk_emb, enc_mask)
+        else:
+            log_durs_predicted = self.duration_predictor(enc_out, enc_mask) 
```

Comment:
We refactored the code in the forward() method but not the same code in the infer() method.

Not necessary here, but ideally we can avoid this by replacing any duplicated logic with a helper function.

Reply:
Makes sense. Done.